File: src/DashboardUI/src/data-contracts/DashboardFilters.ts
import { BeneficialUseListItem } from './BeneficialUseListItem';

export interface DashboardFilters {
  overlays: OverlayFilterSet;
  waterRights: WaterRightsFilterSet;
  timeSeries: TimeSeriesFilterSet;
}

export interface OverlayFilterSet {
  overlayTypes: string[];
  waterSourceTypes: string[];
  states: string[];
}

export interface WaterRightsFilterSet {
  beneficialUses: BeneficialUseListItem[];
  ownerClassifications: string[];
  allocationTypes: string[];
  legalStatuses: string[];
  siteTypes: string[];
  waterSourceTypes: string[];
  states: string[];
  riverBasins: string[];
}

export interface TimeSeriesFilterSet {
  siteTypes: string[];
  primaryUseCategories: string[];
  variableTypes: string[];
  waterSourceTypes: string[];
  states: string[];
}

File: src/API/WesternStatesWater.WestDaat.Common/DataContracts/DashboardFilters.cs
namespace WesternStatesWater.WestDaat.Common.DataContracts
{
    public class DashboardFilters
    {
        public OverlayFilterSet Overlays { get; set; }
        public WaterRightsFilterSet WaterRights { get; set; }
        public TimeSeriesFilterSet TimeSeries { get; set; }
    }

    public class OverlayFilterSet
    {
        public string[] OverlayTypes { get; set; }
        public string[] WaterSourceTypes { get; set; }
        public string[] States { get; set; }
    }

    public class WaterRightsFilterSet
    {
        public BeneficialUseItem[] BeneficialUses { get; set; }
        public string[] OwnerClassifications { get; set; }
        public string[] AllocationTypes { get; set; }
        public string[] LegalStatuses { get; set; }
        public string[] SiteTypes { get; set; }
        public string[] WaterSourceTypes { get; set; }
        public string[] States { get; set; }
        public string[] RiverBasins { get; set; }
    }

    public class TimeSeriesFilterSet
    {
        public string[] SiteTypes { get; set; }
        public string[] PrimaryUseCategories { get; set; }
        public string[] VariableTypes { get; set; }
        public string[] WaterSourceTypes { get; set; }
        public string[] States { get; set; }
    }
}